src/Data/Nat/Binary/Minus.agda

```diff
@@ -0,0 +1,364 @@
+  m = toℕ x;   n = toℕ y;   x≤y = ≤-reflexive x≡y;   m≤n = toℕ-mono-≤ x≤y
+... | tri> _ _ y<x =  begin
+  toℕ (x ∸ y)                    ≡⟨ sym (ℕₚ.m+n∸n≡m (toℕ (x ∸ y)) n) ⟩
+  ((toℕ (x ∸ y)) ℕ.+ n) ℕ.∸ n    ≡⟨ cong (ℕ._∸ n) eq ⟩
+  ((m ℕ.∸ n) ℕ.+ n) ℕ.∸ n        ≡⟨ ℕₚ.m+n∸n≡m (m ℕ.∸ n) n ⟩
+  m ℕ.∸ n                        ∎
+  where
+  open ≡-Reasoning
+  m = toℕ x;   n = toℕ y;   y≤x = <⇒≤ y<x;   n≤m = toℕ-mono-≤ y≤x
+  eq = begin
+    toℕ (x ∸ y) ℕ.+ n    ≡⟨ sym (toℕ-homo-+ (x ∸ y) y) ⟩
+    toℕ ((x ∸ y) + y)    ≡⟨ cong toℕ (+-comm (x ∸ y) y) ⟩
+    toℕ (y + (x ∸ y))    ≡⟨ cong toℕ (x+[y∸x]≡y y≤x) ⟩
+    m                    ≡⟨ sym (ℕₚ.m+[n∸m]≡n n≤m) ⟩
+    n ℕ.+ (m ℕ.∸ n)      ≡⟨ ℕₚ.+-comm n (m ℕ.∸ n) ⟩
+    (m ℕ.∸ n) ℕ.+ n      ∎
+
+∸-magma : Magma 0ℓ 0ℓ
+∸-magma = magma _∸_
+
+import Algebra.Morphism.TwoMagmas ∸-magma ℕₚ.∸-magma
+  as Magmas∸ℕᵇ-ℕ
+fromℕ-homo-∸ :  ∀ m n → fromℕ (m ℕ.∸ n) ≡ (fromℕ m) ∸ (fromℕ n)
+fromℕ-homo-∸ =  Magmas∸ℕᵇ-ℕ.IsHomo⇒IsHomo-rev toℕ fromℕ (cong fromℕ)
+                                       fromℕ-toℕ toℕ-fromℕ toℕ-homo-∸
+
+∸-+ :  ∀ x y o → (x ∸ y) ∸ o ≡ x ∸ (y + o)
+∸-+ x y o =  begin-equality
+  (x ∸ y) ∸ o                     ≡⟨ sym (fromℕ-toℕ ((x ∸ y) ∸ o)) ⟩
+  fromℕ (toℕ ((x ∸ y) ∸ o))       ≡⟨ cong fromℕ (toℕ-homo-∸ (x ∸ y) o) ⟩
+  fromℕ (toℕ (x ∸ y) ℕ.∸ n)       ≡⟨ cong (fromℕ ∘ (ℕ._∸ n)) (toℕ-homo-∸ x y) ⟩
+  fromℕ ((k ℕ.∸ m) ℕ.∸ n)         ≡⟨ cong fromℕ (ℕₚ.∸-+-assoc k m n) ⟩
+  fromℕ (k ℕ.∸ (m ℕ.+ n))         ≡⟨ fromℕ-homo-∸ k (m ℕ.+ n) ⟩
+  fromℕ k ∸ fromℕ (m ℕ.+ n)       ≡⟨ cong ((fromℕ k) ∸_) (fromℕ-homo-+ m n) ⟩
+  fromℕ k ∸ (fromℕ m + fromℕ n)   ≡⟨ cong ((fromℕ k) ∸_) $
+                                     cong₂ _+_ (fromℕ-toℕ y) (fromℕ-toℕ o) ⟩
+  fromℕ k ∸ (y + o)               ≡⟨ cong (_∸ (y + o)) (fromℕ-toℕ x) ⟩
+  x ∸ (y + o)                     ∎
+  where open ≤-Reasoning;  k = toℕ x;  m = toℕ y;  n = toℕ o
+
++-∸ :  ∀ x {y o} → o ≤ y → (x + y) ∸ o ≡ x + (y ∸ o)
++-∸ x {y} {o} o≤y =  begin-equality
+  (x + y) ∸ o                  ≡⟨ sym (fromℕ-toℕ ((x + y) ∸ o)) ⟩
+  fromℕ (toℕ ((x + y) ∸ o))    ≡⟨ cong fromℕ (toℕ-homo-∸ (x + y) o) ⟩
+  fromℕ (toℕ (x + y) ℕ.∸ n)    ≡⟨ cong (fromℕ ∘ (ℕ._∸ n)) (toℕ-homo-+ x y) ⟩
+  fromℕ ((k ℕ.+ m) ℕ.∸ n)      ≡⟨ cong fromℕ (ℕₚ.+-∸-assoc k n≤m) ⟩
+  fromℕ (k ℕ.+ (m ℕ.∸ n))      ≡⟨ fromℕ-homo-+ k (m ℕ.∸ n) ⟩
+  fromℕ k + fromℕ (m ℕ.∸ n)    ≡⟨ cong (_+ (fromℕ (m ℕ.∸ n))) (fromℕ-toℕ x) ⟩
+  x + fromℕ (m ℕ.∸ n)          ≡⟨ cong (x +_) (fromℕ-homo-∸ m n) ⟩
+  x + (fromℕ m ∸ fromℕ n)      ≡⟨ cong (x +_) $ cong₂ _∸_ (fromℕ-toℕ y) (fromℕ-toℕ o) ⟩
+  x + (y ∸ o)                  ∎
+  where
+  open ≤-Reasoning
+  k = toℕ x;  m = toℕ y;  n = toℕ o;  n≤m = toℕ-mono-≤ o≤y
+
+x+y∸x+z :  ∀ x y z →  (x + y) ∸ (x + z) ≡ y ∸ z
+x+y∸x+z x y z =  begin-equality
+  (x + y) ∸ (x + z)    ≡⟨ sym (∸-+ (x + y) x z) ⟩
+  ((x + y) ∸ x) ∸ z    ≡⟨ cong (_∸ z) ([x+y]∸x≡y x y) ⟩
+  y ∸ z                ∎
+  where open ≤-Reasoning
+
+suc-x∸suc-y :  ∀ x y → suc x ∸ suc y ≡ x ∸ y
```

Comment:
Please follow the style guide and use square brackets instead of `-` to indicate application in proof names.

Reply:
Renamed to `suc[x]∸suc[y]`.